code/game/atom/atom_act.dm

```diff
@@ -72,6 +72,13 @@
 	if(!(protection & EMP_PROTECT_WIRES) && istype(wires))
 		wires.emp_pulse()
 
+	if(flags_1 & HOLOGRAM_1) // holograms should ignore EMP_PROTECT flags (since the protection itself is also a hologram)
+		if(prob(75 / severity))
+			// avoid spamming messages for anything inside the holodeck directly
+			if(!istype(get_area(src), /area/station/holodeck))
+				visible_message(span_warning("[src] fades away!"))
+			qdel(src)
```

Comment:
Can you animate the fade out rather than qdeling instantly (`animate(alpha = 0, time = x)` `qdel_in(x)`